dns.go

```diff
@@ -206,6 +206,22 @@ func (c *Client) GetDNSRecord(domainID, name string) (*DNSRecord, error) {
 	return nil, ErrDNSRecordNotFound
 }
 
+// UpdateDNSRecord updates the DNS record
+func (c *Client) UpdateDNSRecord(rc *DNSRecordConfig, d *DNSDomain, r *DNSRecord) (*DNSRecord, error) {
```

Comment:
Why do you need to pass in `*DNSDomain` here? You can get the ID from `r.DNSDomainID` and not need to pass in an unnecessary object.